client/Rules/Formulas/Formula.ts

```diff
@@ -0,0 +1,225 @@
+import { StatBlock } from "../../../common/StatBlock";
+import { IRules } from "../Rules";
+import { AbilityReference } from "./AbilityReference";
+import { Constant } from "./Constant";
+import { Die } from "./Die";
+import { FormulaClass, FormulaResult, FormulaTerm } from "./FormulaTerm";
+import { StatReference } from "./StatReference";
+
+export class Formula implements FormulaTerm {
```

Comment:
I think it would be helpful to rename this and associated things to `DiceFormula`.

Reply:
Fair. I was shying away from specifically mentioning dice since there might be use cases that don't even involve a roll, but 99% of them do and I couldn't come up with any better terms anyway.